### DISCUSSION TITLE
Review the conditions to show the "High FFMPEG CPU usage" message.

### DISCUSSION THREAD

**Discussion by Mincka** [blakeblackshear/frigate#14382] [Ask A Question]
I get the "High FFMPEG CPU usage" showing for a 4K camera with a load of 22%.
Since I have other cameras with lower resolutions, it's "above the average" but not _high_ for my setup.

https://github.com/blakeblackshear/frigate/blob/3f1ab668999d3c27ee77e251200c0f3da81ec22f/web/src/hooks/use-stats.ts#L73-L79

Would it be possible to change the logic for showing this message?

**Comment by NickM-27** [comment]
We can maybe rethink the ranges for ffmpeg, the problem is though we want to make sure users who don't run hwaccel are made aware of the problem

**Reply by Mincka** [reply]
Certainly, and in that case, "High" would be something way over 20% I guess.
https://github.com/blakeblackshear/frigate/blob/eda52a3b8273961db60d367c25f552522c514bd6/web/src/types/graph.ts#L46-L49

An alternative could be the ability to override these values in the configuration file.

**Reply by NickM-27** [reply]
not necessarily, ffmpeg usage can be ~ 20% on my server for example but with hwaccel it is 2%. Will have to see what makes sense

**Reply by Mincka** [reply]
So the value can be anything, you won't be able to tell if it's good or bad.
Also, you cannot predict what will be the load difference with / without hardware acceleration. 

Aren't you able to tell if hardware acceleration is used or not?

![image](https://github.com/user-attachments/assets/569ed23e-1ea3-42e5-b7a3-abcfb5bd2982)

It could be an additional condition to avoid displaying this message or adjust the threshold value in that case.

**Reply by NickM-27** [reply]
in general, 20% is quite high for ffmpeg with hwaccel and usually indicates an issue or something that can be optimized. This message is there to alert people and encourage them to seek support or fix the issue. Not sure if we will want to change this, but in any case like I said will think about what might make sense

**Comment by mbrinkley1234** [comment]
Any chance this is still being reviewed?  I get this error frequently because I am using a high quality h265 feed for detect because I want to be able to detect objects that are far away (small).  I am aware FFMPEG is high, and I dont need a reminder.  I think the ability to adjust the threshold would be best, but an ignore until next reboot would be helpful.  


![Screenshot 2025-04-29 095419](https://github.com/user-attachments/assets/26f65eca-8718-4b26-a357-d2e5bf30cd69)

**Reply by NickM-27** [reply]
Adjusting the threshold would be a messy solution, as there are many different thresholds and presumably all would need to be configurable. We are still thinking through a solution that might make sense. Even if a user is using hardware acceleration, it is difficult to know if the high CPU usage is "acceptable" for the user, or if they don't realize that there is a problem because they didn't realize using a high res stream is generally not recommended, or perhaps they are using the incorrect hwaccel preset (vaapi vs qsv for example)

**Reply by mbrinkley1234** [reply]
I dont know the behind the scenes coding required, but it seems like if I could add a line to my config for a specific camera that says 'ffmpeg CPU usage threshold: 45' and then just not have the alert show up until its above 45% for that camera seems like it would make sense.  Requiring a user to add it per camera seems like it would weed out the 'something is wrong and I didn't realize it' scenarios, and would also keep the error intact for cameras that should not be running high.

**Reply by NickM-27** [reply]
yes, but that would also mean we have an additional 6 configuration items that are configurable per camera, along with ~20 other thresholds that are global only. That is a lot of config for us to add, and then have to maintain as we change the UI and potentially adjust the threshold behavior, leading to requiring more configuration migrations. So that is the point, IMO it is best to avoid user configurable thresholds, and find a better solution. Especially as this will increase the effort when we want to add additional thresholds in the future.

**Reply by mbrinkley1234** [reply]
That makes sense.  Thanks for the insight.

Thanks
